src/lib.rs

```diff
@@ -923,6 +940,28 @@ mod tests {
             _ => panic!("expected csi sequence"),
         }
     }
+
+    #[cfg(feature = "no_std")]
+    #[test]
+    fn build_with_fixed_size() {
+        static INPUT: &[u8] = b"\x1b[3;1\x1b[?1049h";
+        let mut dispatcher = Dispatcher::default();
+        let mut parser: Parser<30> = Parser::new_with_size();
+
+        for byte in INPUT {
+            parser.advance(&mut dispatcher, *byte);
+        }
+
+        assert_eq!(dispatcher.dispatched.len(), 1);
+        match &dispatcher.dispatched[0] {
+            Sequence::Csi(params, intermediates, ignore, _) => {
+                assert_eq!(intermediates, &[b'?']);
+                assert_eq!(params, &[[1049]]);
+                assert!(!ignore);
+            },
+            _ => panic!("expected csi sequence"),
+        }
+    }
 }
```

Comment:
It would probably make sense to test this with OSC escapes. Ideally even to check that it breaks when the limit is exceeded.

Reply:
Added some tests. Apparently the behaviour is to silently truncate when the limit is exceeded - unsure if that's intended or not.

Reply:
It is. The limit is supposed to be the upper limit of what you'll ever use. If that isn't big enough for your purposes, that's on you for setting a limit that is too small.